src/runner.js

```diff
@@ -41,74 +55,180 @@ TODO?  Promises or callback as first arg?  Are we node?
 
 let actions = require("./common/actions");
 let chai = require("chai");
+let type = require("./jetpack/type");
+
+const config = {
+  alwaysRun: false
+};
 
-let config = exports.config = {};
+
+let validate = function (thing, rules) {
```

Comment:
use chai here is overkill.  pick a contracts / validation system somewhere and use it everywhere.